Add a new compute engine channel creds interop test case

apolcyn opened the pull request:
Adds a new interop test case to exercise implementations of new "compute engine channel creds" API's in cloud to prod.

Also does some cleanup (gets rid of the "google_default_credentials" option to `--transport_security` arg, because it's unused).

This needs to wait for https://github.com/grpc/grpc-java/pull/5478 before it will start passing.

jiangtaoli2016@ can you please review the description of the new interop test?

jtattermusch commented on tools/run_tests/run_interop_tests.py:
```diff
@@ -800,9 +851,15 @@ def cloud_to_prod_jobspec(language,
         transport_security_options = [
             '--custom_credentials_type=google_default_credentials'
         ]
+    elif transport_security == 'compute_engine_channel_creds' and str(
```
the previous elif branch is `transport_security == 'google_default_credentials'` but you're removing that option.

apolcyn replied:
This was a bit weird. I've removed `google_default_credentials` as an option for a value of `--transport_security` argument to the whole script, but it's still a possible value for the parameter to `cloud_to_prod_jobspec`, because we're still appending "cloud_to_prod_jobspec with google default credentials" [to the list of test cases](https://github.com/grpc/grpc/blob/master/tools/run_tests/run_interop_tests.py#L1408).

apolcyn commented:
squashed down commits and added `--default_service_account` arg to this test (to fix it) - https://github.com/grpc/grpc/pull/18370/files#diff-267afa22a65534e2688adc5fce0f3f47R850

apolcyn commented:
TSAN C++ failure: https://github.com/grpc/grpc/issues/18142
